snuba/query/parser/__init__.py

```diff
@@ -128,16 +136,18 @@ def build_selected_expressions(
     for orderby in to_list(body.get("orderby", [])):
         if isinstance(orderby, str):
             match = NEGATE_RE.match(orderby)
-            assert match is not None, f"Invalid Order By clause {orderby}"
+            if match is None:
+                raise ParsingException(f"Invalid Order By clause {orderby}")
             direction, col = match.groups()
             orderby = col
         elif is_function(orderby):
             match = NEGATE_RE.match(orderby[0])
-            assert match is not None, f"Invalid Order By clause {orderby}"
+            if match is None:
+                raise ParsingException(f"Invalid Order By clause {orderby}")
             direction, col = match.groups()
             orderby = [col] + orderby[1:]
         else:
-            raise ValueError(f"Invalid Order By clause {orderby}")
+            raise ParsingException(f"Invalid Order By clause {orderby}")
```

Comment:
These error messages don't really do a great job explaining what is invalid about the clause/how to fix it.